js/twitter-text.js

```diff
@@ -260,7 +260,7 @@
   twttr.txt.regexen.urlHasHttps = /^https:\/\//i;
 
   // cashtag related regex
-  twttr.txt.regexen.cashtag = /[a-z]{1,6}(?:[._][a-z]{1,2})?/i;
+  twttr.txt.regexen.cashtag = /(?:[a-z]{1,2}[:._-]{1}[0-9]{1,6})|(?:[a-z]{1,6}(?:[:._-]{1}(?:[a-z]|[0-9]){1,6})?)|(?:[0-9]{1,6}[:._-]{1}[a-z]{1,2})/i;
```

Comment:
1. Removed those. Thanks! 
2. we're using /i at the end of each regex for case insensitivity in js/rb/java. option NSRegularExpressionCaseInsensitive in objc.